lib/bloc/user_profile/user_profile_bloc.dart

```diff
@@ -1,33 +1,80 @@
 import 'dart:async';
 import 'dart:io';
 
+import 'package:breez_translations/breez_translations_locales.dart';
+import 'package:c_breez/bloc/user_profile/default_profile_generator.dart';
 import 'package:c_breez/bloc/user_profile/user_profile_state.dart';
 import 'package:c_breez/models/user_profile.dart';
 import 'package:c_breez/services/breez_server.dart';
 import 'package:c_breez/services/notifications.dart';
+import 'package:fimber/fimber.dart';
 import 'package:hydrated_bloc/hydrated_bloc.dart';
 import 'package:path_provider/path_provider.dart';
 
 const PROFILE_DATA_FOLDER_PATH = "profile";
 
+final _log = FimberLog("UserProfileBloc");
+
 class UserProfileBloc extends Cubit<UserProfileState> with HydratedMixin {
   final BreezServer _breezServer;
   final Notifications _notifications;
 
-  UserProfileBloc(this._breezServer, this._notifications)
-      : super(UserProfileState.initial());
+  UserProfileBloc(
+    this._breezServer,
+    this._notifications,
+  ) : super(UserProfileState.initial()) {
+    var profile = state;
+    _log.v("State: ${profile.profileSettings.toJson()}");
+    final settings = profile.profileSettings;
+    if (settings.color == null || settings.animal == null) {
```

Comment:
It looks like if the color and animal are null then name is null as well so maybe we can generate the name inside this block as well removing the bellow block?

Reply:
I'm a little bit paranoic, but I want to avoid overriding anything that exists by doing it on the same block I need to use the actual.color ?? new.color and create a new generator, having in separated blocks, I can guarantee no property override at all.